### PR TITLE
add support for fish shell

### PR DESCRIPTION
**Pull request by Baisang**
seems to work

### CODE REVIEW

File: modules/ocf/files/shell/config.fish [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Quit if not interactive
+if not status --is-interactive
+    exit
+end
```
Comment (chriskuehl) [inline modules/ocf/files/shell/config.fish]: i don't know anything about fish, but if exit means the same thing as it does in sh, i wonder if this causes the entire shell to exit? lol

Comment (Baisang) [reply modules/ocf/files/shell/config.fish]: oh shit you're right lmao

Comment (Baisang) [reply modules/ocf/files/shell/config.fish]: oh wait https://fishshell.com/docs/current/commands.html#exit

`If exit is called while sourcing a file (using the source builtin) the rest of the file will be skipped, but the shell itself will not exit.`

Does that seem to be intended behavior?

Comment (chriskuehl) [reply modules/ocf/files/shell/config.fish]: yeah, that sounds right (different than the sh behavior, but that's ok)

File: modules/ocf/manifests/packages/shell.pp [hunk 2]
```diff
@@ -1,5 +1,5 @@
 class ocf::packages::shell {
-  package { ['bash', 'tcsh', 'zsh']: }
+  package { ['bash', 'tcsh', 'zsh', 'fish']: }
```
Comment (abizer) [inline modules/ocf/manifests/packages/shell.pp]: fish

**Comment by matthew-mcallister** [comment]
Duplicate this in [ocflib](https://github.com/ocf/ocflib/blob/master/ocflib/misc/validators.py#L45-L54)?

**Comment by redplanks** [comment]
So long and thanks for all the zsh

**Pull request merged**